lib/cockpit-components-select.jsx

```diff
@@ -85,26 +84,20 @@
         },
         render: function() {
             var self = this;
-            var currentValue = this.state.currentValue;
-            if (currentValue === undefined && 'initial' in this.props)
-                currentValue = this.props.initial;
+            var currentValue;
 
-            var listItems;
-            if (this.props.children) {
-                listItems = this.props.children.map(function(itm) {
-                    var data = ('data' in itm.props)?itm.props.data:undefined;
-                    // we need to have some kind of value
-                    var value = (itm.props.children !== undefined)?itm.props.children:textForUndefined;
-                    // if we don't have anything selected, take the first item
-                    if (currentValue === undefined) {
-                        currentValue = value;
-                        self.setState({ currentValue: currentValue, currentData: data });
-                        self.props.onChange(data);
-                    }
-                    return <li data-value={value} data-data={data}>{itm}</li>;
-                });
-            }
-            var classes = "btn-group bootstrap-select dropdown form-control";
+            var listItems = React.Children.map(this.props.children, function(itm) {
+                var data = ('data' in itm.props)?itm.props.data:undefined;
```

Comment:
I agree, but I only reindented those lines and didn't want to change the style, because it's like that in other parts of this file as well. Fixed all occurrences now.